pyproject.toml

```diff
@@ -372,6 +376,7 @@ module = [
     "sentry.integrations.bitbucket_server.client",
     "sentry.integrations.bitbucket_server.integration",
     "sentry.integrations.bitbucket_server.webhook",
+    "sentry.integrations.discord.utils.channel",
```

Comment:
the allowlist should not be increased -- we're trying to _improve_ type checking not regress it

Reply:
@asottile-sentry i'm trying to remove it. I'm unsure why it's adding it. 

`mypy` passes locally for me.

Reply:
lemme check it out and see what's going on here...

Reply:
@asottile-sentry I think fix might be https://github.com/getsentry/sentry/pull/61655/commits/856cec1e7d308a466976fe4354d6ea7a297efe78. I'm unsure why `mypy` didn't complain for me locally 🤔

Reply:
ah I found it --

```
$ grep utils/channel .artifacts/mypy-all
src/sentry/integrations/slack/utils/channel.py:133: error: Argument 2 to "check_for_channel" has incompatible type "Optional[str]"; expected "str"  [arg-type]
src/sentry/integrations/slack/utils/channel.py:202: error: Unsupported right operand type for in ("Union[BaseApiResponse, Mapping[str, Any], Response]")  [operator]
src/sentry/integrations/slack/utils/channel.py:207: error: Value of type "Union[BaseApiResponse, Mapping[str, Any], Response]" is not indexable  [index]
src/sentry/integrations/slack/utils/channel.py:208: error: Value of type "Union[BaseApiResponse, Mapping[str, Any], Response]" is not indexable  [index]
src/sentry/integrations/slack/utils/channel.py:213: error: Value of type "Union[BaseApiResponse, Mapping[str, Any], Response]" is not indexable  [index]
src/sentry/integrations/discord/utils/channel.py:38: note: "validate_channel_id" defined here
```

the allowlist-thingy added `utils.channel` due to the `note:` -- the error was actually coming from a different error though! (which is allowlisted at the moment)

Reply:
Oh interesting. Thanks for checking!